### DISCUSSION TITLE
why is the dev server generating .gitignore

### DISCUSSION THREAD

**Discussion by juliusmarminge** [expo/router#846] [General]
### Which package manager are you using? (Yarn is recommended)

pnpm (known issues due to Metro)

### Summary

I enabled the experimental typedRoutes features and ever since expo router is generating a `.gitignore` everytime i run dev.

I think it's kinda silly to generate this file multiple times, I tend to not have nested gitignores in my monorepos but just keep a general purpose one in root, so having expo-router generate one for me everytime i run dev is kinda annoying

---

I think it's fair to assume devs can put things in their gitignores themselves, and you don't have to generate it for us.

### Minimal reproducible example

Happy to provide one if you deem necessary, for now just check out [create-t3-turbo](https://github.com/t3-oss/create-t3-turbo/tree/main/apps/expo) where i have the gitignore in root and don't want one in the expo directory

**Comment by marklawlor** [comment]
There's two reasons:

The practical reason is that features are still experimental and they are not as strictly covered by semver. By forcing them to be inside a `gitignore` it gives us more flexibility to make changes without forcing the user to commit updates seemingly randomly updates.

The stylist reason, `gitignore` is designed to be nested/inherited and choosing not to do so is a (valid) personal choice. Mono-repo file layouts are still a topic of discussion, but I'm observing that the trend seems to be moving towards having a top level Project configs with App level configs (that inherit from Project). As `gitignore` was designed to be nested, and the config we are generating is App specific, I believe it makes sense that the config we are generating should live at the app level (we can provide a flag to disable this behaviour).

I'm going to close this issue and move it to a discussion as I am interested in people's opinions on the subject.

**Reply by juliusmarminge** [reply]
Makes sense.

I'm thorn between having a root gitignore vs project level ones. I already feel like there are too many config levels in a javascript project so if i can reduce that by moving certain parts into a single file, I'm happy to do so. But I also see the case for keeping the individual app/package independent and ignore only what it needs.

But if you could provide a flag that'd be nice. However I will say expo is probably the first time I've seen a gitignore generated during watch.

**Comment by juliusmarminge** [comment]
Hey @marklawlor,

It seems like this is now causing issues due to lack of permissions on windows (see https://github.com/t3-oss/create-t3-turbo/issues/873). Any chance we can get an override option to skip generation?

**Reply by marklawlor** [reply]
If I understand the issue correctly, Expo CLI is running with different permissions than the process that created the project folder. Even if we added a flag to skip `.gitignore`, isn't it just going to fail when we generate the type files? And if it doesn't it's probably because Expo created a new folder with different permissions to it's parent folder - which isn't great either 🫤

This also means other file generation commands like `expo prebuild` are potentially not working. 

I think we need to address the permission issue rather than workaround this single case

**Reply by juliusmarminge** [reply]
Do you want me to open a new issue related to the permission issue?

**Comment by dBianchii** [comment]
+1
